src/main/java/org/testng/internal/reflect/ReflectionHelper.java

```diff
@@ -38,6 +38,36 @@
     return result;
   }
 
+  /**
+   * @return An array of all locally declared methods or equivalent thereof
+   * (such as default methods on Java 8 based interfaces that the given class
+   * implements) but excludes the <code>main()</code> method alone.
+   */
+  public static Method[] excludingMain(Class<?> clazz) {
+    Method[] declaredMethods = clazz.getDeclaredMethods();
+    List<Method> pruned = new LinkedList<>();
+    for (Method declaredMethod :declaredMethods) {
+      if ("main".equals(declaredMethod.getName()) && isStaticVoid(declaredMethod) && acceptsStringArray(declaredMethod)) {
+        continue;
```

Comment:
@juherr - I think Earlier it was never possible to include a `main()` method as part of the execution chain. So this warning is not going to add any value.